src/systems/bit-media-frames.js

```diff
@@ -229,10 +316,16 @@ function mediaTypesOf(world, entities) {
   return mask;
 }
 
+export function cleanupMediaFrame(obj) {
+  obj.traverse(child => {
```

Comment:
No need for the wrapping function.

That said not sure this is correct. The only thing the media frame "owns" in terms of disposable resources is the geometry for the guide. The cloned preview object is all just references to geometries/textures that something else owns. Its actually incorrect to try and `.dispose()` a geometry/texture for an object we happened to be previewing just because this media frame is going away, since the object that actually "owns" those things is not going away. 

With the way threejs resource management works you won't actually notice this issue since it will lazily re-create everything. So I think what will happen in practice is when you delete a media frame capturing an object we will delete the GPU side resources for them and then re-upload them to the GPU next frame.

Reply:
Oof, actually looking at the code some more, for videos we actually are creating a new Geometry so we technically do want to clean that up. We actually don't need a new geometry for each media frame and I am fine with statically keeping that around for the lifetime of the application so we shouldn't need to clean that up. Will add a note above about that as well.

Reply:
This should only be disposing the guide as the preview is transient as is removed when hidden so it shouldn't be affected by the cleanup. But you are right, the video geometry is not being cleaned up correctly. I didn't want to preallocate that to avoid allocating resources for something that can potentially never be used but maybe that's better that disposing it every time a preview is hidden so I'm going o do that.